### PR TITLE
Update python version 3.8 -> 3.11

### PR DESCRIPTION
**Pull request by jkgoodrich**
Suggested for [Hail](https://atgu.slack.com/archives/C055FKCJHAB/p1691438191032509)

Also added `gnomad` as a known third party to keep `isort` from changing the spacing

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,10 +13,10 @@ jobs:
         uses: actions/checkout@v2
         with:
           submodules: recursive
-      - name: Setup Python 3.8
+      - name: Setup Python 3.11
         uses: actions/setup-python@v2
```
Comment (ch-kr) [inline .github/workflows/ci.yml:17]:
```suggestion
        uses: actions/setup-python@v4
```
^I think this is the current version if we want to update; https://github.com/actions/setup-python

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,10 +13,10 @@ jobs:
...
         with:
-          python-version: 3.8
+          python-version: 3.11
       - name: Use pip cache
         uses: actions/cache@v2
```
Comment (ch-kr) [inline .github/workflows/ci.yml:21]: I think this is v3 now as well: https://github.com/actions/cache

**Review by ch-kr — APPROVED** [review]
some minor comments asking if we want to update tool versions, but LGTM!

File: .pre-commit-config.yaml
Comment (ch-kr) [inline .pre-commit-config.yaml:1]: as a general question for the future, should we exclude v2 code from checks? we aren't actively using it, so we shouldn't continually have to maintain it. what are your thoughts? this also came up in this PR (https://github.com/broadinstitute/gnomad_methods/pull/576); v2 still uses the deprecated `hl.cond`

**Review by ch-kr — APPROVED** [review]
one non-urgent question for the future, but LGTM!

also noting that I only quickly looked over the files reformatted by black

File: .pre-commit-config.yaml
Comment (jkgoodrich) [reply .pre-commit-config.yaml:1]: Yeah, we do for some of them that require more than just a quick run to reformat ex: `pydocstyle --match-dir='(?!v2|cuKING).*' gnomad_qc` 
I was going to tell @KoalaQin at the meeting today that we don't need to go back and update `hl.cond` in past versions

**Pull request merged**